<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"><html><head>	<title>4Peaks Help</title>	<link rel="StyleSheet" type="text/css" href="style.css">	<meta name="AppleIcon" content="../gfx/icon_32x32.png">	<meta name="Copyright" content="Copyright 2004-2015 Nucleobytes, All Rights Reserved."></head><body bgcolor="#ffffff" link="#666666" vlink="#666666">	<div id="background-logo" class="center"></div>	<table border="0" cellpadding="1" cellspacing="0" width="100%" class="menu">		<tr>			<td><center><a href="quickstart.html"><font size="2">Quick start</font></a></center></td>			<td><center><a href="main.html"><font size="2">Main Window</font></a></center></td>			<td><center><a href="file.html"><font size="2">File Menu</font></a></center></td>			<td><center><a href="edit.html"><font size="2">Edit Menu</font></a></center></td>			<td><center><a href="plugins.html"><font size="2">Plugins</font></a></center></td>		</tr>		<tr>			<td class="current"><center><a href="sequencing.html"><font size="2">Sequencing</font></a></center></td>			<td><center><a href="troubles.html"><font size="2">Troubleshooting</font></a></center></td>			<td><center><a href="history.html"><font size="2">Sequencing History</font></a></center></td>			<td><center><a href="nucleobytes.html"><font size="2">About Nucleobytes</font></a></center></td>			<td></td>		</tr>	</table>		<br>		<div class="page-header">	<table border="0" cellpadding="1" cellspacing="0" width="100%">		<tr height="40">			<td width="40" height="40"><img height="40" width="40" src="../gfx/icon_64x64.png"></td>			<td width="6" height="40"></td>			<td height="40" valign="middle">				<font size="4" color="#ff71a5">DNA Sequencing Techniques</font>			</td>		</tr>	</table>	</div>		<br>	<!-- AppleSegStart="4Peaks Help Index" -->	<!-- AppleSegDescription="4Peaks Help Start Page" -->		<!-- AppleKeywords="sequencing, PCR, polymerase, chain, reaction, fluorescent, probes, dideoxy, nucleotides, shotgun, shotgunning, shot, walking, primers, primer" -->	<p class="main-text">In genetics, DNA sequencing is a complex nucleotide-sequencing technique including three identifiable steps:<br>		<ol>				<li><a href="#pcr">Polymerase Chain Reaction (PCR)</a></li>			<br>			<li><a href="#reaction">Sequencing Reaction</a></li>			<br>			<li><a href="#gel">Gel Electrophoresis &amp; Computer Processing</a></li>		</ol>	</p>	<br>		<span class="section-header"><a name="intro" id="intro"><font color="#FF71A5">Introduction</font></a></span>	<br>		<p class="main-text">		Gene therapy is a process that changes the nucleotide bases in DNA, thereby bringing out a desired effect in the organisms receiving treatment. By manipulating the base nucleotides, researchers change the genes and consequently the proteins they produce. Since most proteins achieve their function by folding into their natural shape, if the sequence of amino acids changes, then the shape of the proteins change. Once the proteins change, the active site of the protein either becomes inactive or performs a slightly different function. Scientists alter the sequence of DNA, or insert new segments of genetic code to replace old ones, in order to achieve these effects.	</p>		<br>		<span class="section-header"><a name="pcr" id="pcr"><font color="#FF71A5">Polymerase Chain Reaction (PCR)</font></a></span>	<br>		<p class="main-text">		PCR is described as a vector-based cloning procedure used to generate many copies of DNA from an initially small sample. That consequently mirrors the way in which DNA is replicated within a cell. PCR is important when sequencing DNA and numerous other genetic processes, because it creates millions of copies of a single DNA strand or fragment, which lets scientists perform thousands of reactions on the same string of DNA, reducing the number of variable differences found in organisms on the genetic level. For example, when an institute wishes to research skin cells of an individual, rather than removing ten million or so cells, they instead take a few thousand or even a few hundred and isolate the desired DNA, which they then PCR. The result is several million strings of perfectly alike DNA.<br>		<br>		<ul>				<li>Step 1: Scientists isolate a piece of DNA to be amplified (copied) and then heat it in a test tube, which causes the amino acid bonds in the double helix to break apart (two complementary halves).</li>			<br>			<li>Step 2: As the solution of DNA halves cools, short pieces of DNA called primers are added into the test tubes. The primers attach the halves, starting the segment that will be cloned.</li>			<br>			<li>Step 3: Free-floating nucleotides (the letters that make up DNA) and an enzyme called DNA polymerase are then added to the solution. The polymerase enzyme causes the nucleotides to bond to the primer and the rest of the DNA string until the segment of DNA is copied on both halves making two identical strands of DNA.</li>			<br>		</ul>	</p>	<br>		<center>	<div class="screenshot">		<img src="../gfx/pcrmix.gif"><br>		<p>PCR Reaction Mix</p>	<div>	</center>		<br>			<p class="main-text">		Each cooling and heating cycle causes the amount of DNA segments in the test tube to double. Encarta explains, 		'In a matter of hours, scientists can obtain millions of copies of a desired piece of DNA. PCR enables scientists to amplify 		traces of DNA found at a crime scene or in a fossil of animal to produce sufficient quantities to study (How Scientists Work 		with Genes, 2000).'	</p>		<br>	<span class="section-header"><a name="reaction" id="reaction"><font color="#FF71A5">Sequencing Reaction</font></a></span>	<br>		<p class="main-text">		During this process the nucleotide deoxyribonucleic acid triphosphates (dNTPs) and one dideoxyribonucleic acid 		triphosphate (ddNTP) are added to rebuild the DNA fragment from the PCR reaction; both nucleotides closely resembles 		one of the DNA letters (A, T, G, or C). Presently there is two widely used sequencing methods:<br>		<ul>				<li>The Conventional Method</li>			<br>			<li>Shotgunning</li>			<br>		</ul>	</p>		<p class="main-text">		Once scientists use PCR to create many copies of a single strand of the DNA fragment they begin to synthesis the location of 		each letter. The conventional method involves the following:<br>		<ul>				<li>Step 1: Place identical DNA strands into four test tubes, each one containing addNTP that resembles one of the four nucleotides in DNA (A, T, C, G) and lots ofdNTPs (which are free-floating nucleotides) that also resemble the DNA letter,  except they do not build functioning DNA chains.</li>			<br>			<li>Step 2: Then add the polymerase enzyme and a known primer DNA, similar to the one used in PCR, to the test tubes.	 The primer marks the beginning of each sequenced string of DNA. In each test tube, the dNTPs, which act like letters in	 a DNA bond with the complementary nucleotide, thereby copying the original strand. However, the ddNTP in each test 	 tube also bond with the DNA fragments at a probable ratio of 1 bond 100 times it could bond. Each time this happens 	 the copy terminates, thereby creating millions of DNA strings of differing length that start with the same primer and 	 each ending with the same ddNTP nucleotide; determined by which of the four test tube being analyzed, since only one 	 of the ddNTPs are in each test tube.</li>			<br>			<li>Step 3: Then use Gel Electrophoresis to arrange the DNA pieces from largest to smallest and X-ray detection to 	determine the length of the strings.</li>		</ul>	</p>		<br>		<center>	<div class="screenshot">		<img src="../gfx/cycle.gif"><br>		<p>The PCR cycle</p>	<div>	</center>		<br>		<p class="main-text">		This is the method that was first developed. Fortunately, contemporary institutes no longer use this exact method but one 		that is four times faster. By using ddNTPs tagged with fluorescents they no longer need four test tubes, but a single one, 		which contains all four fluorescent ddNTPs. They then rely on computers to detect the different colors of the ending pieces 		of the DNA segments after Gel Electrophoresis to determine the letters and length of the DNA chain.	</p>		<br>		<center>	<div class="screenshot">		<img src="../gfx/sangers.gif">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<img src="../gfx/modern.gif"><br>		<p>Radioactive <i>vs</i> Fluorescent sequencing method.</p>	<div>	</center>		<br>		<span class="section-header"><a name="speech" id="speech"><font color="#FF71A5">Shotgun Sequencing</font></a></span>	<br>		<p class="main-text">		The 'shotgunning' technique is a faster method of sequencing that breaks the original strands into segments of differing length, which are then detected and ordered by computers.		<br>		<ul>				<li>Step 1: Blend the post PCR DNA string that is to be sequenced into little fragments.</li>			<br>			<li>Step 2: Place the new segments into a test tube filled with the polymerase enzyme and a primer bit of DNA.</li>			<br>			<li>Step 3: Now, like in the original method, let the DNA rebuilds itself with the dNTPs and the fluorescent ddNTPs tagged in the test tubes until all the strings have terminated.</li>			<br>			<li>Step 4: Use Gel Electrophoresis to sort the fragments by size and a computer to record the many DNA fragments lengths. Lastly the computer should process and realign these fragments into the original string, thereby sequencing it.</li>		</ul>	</p>		<br>		<p class="main-text">		The advantage of using smaller fragments of the larger DNA chain is that since the time required to sequence the DNA has been greatly shorted. Therefore, machines can sequence the fragments many times in order to achieve a high level of accuracy, by using sequencing software which lines up the DNA by finding overlapping letter sequences in the many pieces after the gel electrophoresis. However, scientists have experienced a few problem when 'shotgunning' DNA string with many common and reoccurring sequences; therefore researchers using the 'shotgunning' process often sequence the DNA both backwards and forwards to return more accurate results. The following graphic shows how the small DNA fragments are realigned to assemble the full sequence.<br>		<br>	</p>	<br>		<center>	<div class="screenshot">		<img src="../gfx/shotgun.gif"><br>		<p>Shotgun sequencing technique</p>	<div>	</center>		<br>		<p class="main-text">		The DNA fragments are lined up. In actuality billions of overlapping DNA pieces need to be aligned for 		an acceptable accuracy. Fortunately, by highly automating the 'shotgunning' method of sequencing, scientists are quickly 		organizing enough DNA pieces to return precise chains faster than competitors using more conventional techniques.<br>	</p>		<br>		<span class="section-header"><a name="gel" id="gel"><font color="#FF71A5">Gel Electrophoresis &amp; Computer Processing</font></a></span>	<br>		<p class="main-text">		Solutions containing the DNA fragments are placed within a thick gel. Then a mild electric current is applied to the gel, 		causing one end of the gel to have a positive charge and the other to have a negative charge. All of the fragments begin to 		move from the negative end of the gel towards the positive end, except the short strings move faster than the larger ones. 		Thus several hours later when the power is terminated the DNA becomes spread out over the gel with the smaller strands 		closer to the positive end. This creates a stair-like effect with the short DNA fragments on the positive side increasing 		gradually to the longer pieces on the negative side of the gel. Each string ends with a ddNTP, which researchers detect 		with x-rays or more commonly fluorescent detectors, before using computers to organize the results.<br>		<br>		The computer starts processing by first detecting the DNA primer fragments (known sequences of nucleotides starting 		each fragment). Next, the computer determines the letter of the nucleotide following the primer (the first location); 		the computer continues to move up the stairs, detecting the letter at each location. After this is completed, the program 		simply returns the letter value at each rung of the DNA ladder and scientists ascertain the sequence.<br>	</p>		<br>		<p class="main-text">		Text adapted from a document originally published at: <a href="http://www.uweb.ucsb.edu/~trevorc/">http://www.uweb.ucsb.edu/~trevorc/</a><br>	</p>		<br>		<table border="0" width="100%">		<tr> 			<td bgcolor="#c0c0c0" height="3"></td>		</tr>	</table>	<br>	<table border="0" cellpadding="1" cellspacing="0" width="100%" class="menu">		<tr>			<td><center><a href="quickstart.html"><font size="2">Quick start</font></a></center></td>			<td><center><a href="main.html"><font size="2">Main Window</font></a></center></td>			<td><center><a href="file.html"><font size="2">File Menu</font></a></center></td>			<td><center><a href="edit.html"><font size="2">Edit Menu</font></a></center></td>			<td><center><a href="plugins.html"><font size="2">Plugins</font></a></center></td>		</tr>		<tr>			<td class="current"><center><a href="sequencing.html"><font size="2">Sequencing</font></a></center></td>			<td><center><a href="troubles.html"><font size="2">Troubleshooting</font></a></center></td>			<td><center><a href="history.html"><font size="2">Sequencing History</font></a></center></td>			<td><center><a href="nucleobytes.html"><font size="2">About Nucleobytes</font></a></center></td>			<td></td>		</tr>	</table>		<br><!-- AppleSegEnd --></body></html>