<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"><html><head>	<title>4Peaks Help</title>	<link rel="StyleSheet" type="text/css" href="style.css">	<meta name="AppleIcon" content="../gfx/icon_32x32.png">	<meta name="Copyright" content="Copyright 2004-2015 Nucleobytes, All Rights Reserved."></head><body bgcolor="#ffffff" link="#666666" vlink="#666666">	<div id="background-logo" class="center"></div>	<table border="0" cellpadding="1" cellspacing="0" width="100%" class="menu">		<tr>			<td><center><a href="quickstart.html"><font size="2">Quick start</font></a></center></td>			<td><center><a href="main.html"><font size="2">Main Window</font></a></center></td>			<td><center><a href="file.html"><font size="2">File Menu</font></a></center></td>			<td><center><a href="edit.html"><font size="2">Edit Menu</font></a></center></td>			<td><center><a href="plugins.html"><font size="2">Plugins</font></a></center></td>		</tr>		<tr>			<td><center><a href="sequencing.html"><font size="2">Sequencing</font></a></center></td>			<td class="current"><center><a href="troubles.html"><font size="2">Troubleshooting</font></a></center></td>			<td><center><a href="history.html"><font size="2">Sequencing History</font></a></center></td>			<td><center><a href="nucleobytes.html"><font size="2">About Nucleobytes</font></a></center></td>			<td></td>		</tr>	</table>		<br>		<div class="page-header">	<table border="0" cellpadding="1" cellspacing="0" width="100%">		<tr height="40">			<td width="40" height="40"><img height="40" width="40" src="../gfx/icon_64x64.png"></td>			<td width="6" height="40"></td>			<td height="40" valign="middle">				<font size="4" color="#ff71a5">Troubleshooting For DNA Sequencing</font>			</td>		</tr>	</table>	</div>		<br>	<!-- AppleSegStart="4Peaks Help Index" -->	<!-- AppleSegDescription="This section is about troubleshooting. With tips and tricks for better sequencing results." -->		<!-- AppleKeywords="problems, peaks, no peaks, high, troubles, troubleshoting, " -->		<p class="main-text">Automated fluorescent sequencing is much more sensitive to quality and quantity factors than manual sequencing procedures. There are multiple variables that can affect data quality obtained from automated sequencing. This troubleshooting guide is designed to help you create solutions for your problematic DNA sequencing data. This page describes the following topics:<br>		<ol>				<li><a href="#practical">Practical Examples of Sequencing Data</a></li>			<br>			<li><a href="#reactions">Troubleshooting DNA Sequencing Reactions</a></li>			<br>			<li><a href="#composition">Troubleshooting DNA Sequence Composition Problems</a></li>		</ol>	</p>	<br>		<span class="section-header"><a name="practical" id="practical"><font color="#FF71A5">Practical Examples Of Sequence Data</font></a></span>	<br>		<p class="main-text"><b>High-quality data.</b><br><br>The following are common features of high quality sequencing data (see Figure 1):<br>		<ul>				<li>Peaks should be well defined</li>			<li>There should be accurate base calling with few or no Ns</li>			<li>Peak height should gradually decrease as the base count increases</li>			<li>There should be little or no background noise in the data</li>		</ul>	</p>		<br>		<center>	<div class="screenshot">		<img src="../gfx/goodsequence.png"><br>		<p>Figure 1. Example of a good sequence result in 4Peaks.</p>	<div>	</center>		<br>		<p class="main-text"><b>No Recognizable Sequence.</b><br><br>The following are common features of poor quality sequencing data (see Figure 2):<br>		<ul>				<li>Few base calls and numerous noise.</li>			<li>High noise levels.</li>			<li>Early signal loss.</li>		</ul>	</p>		<br>		<center>	<div class="screenshot">		<img src="../gfx/wrongprimer.png"><br>		<p>Figure 2. Example of a sequence with the wrong primer in 4Peaks.</p>	<div>	</center>		<br>		<p class="main-text">Figure 2 above illustrates analyzed data with a high level of noise and no well defined peaks. This particular reaction failed  due to the absence of a priming site in the template for the primer that was used. If excess dye peaks are present, they can raise signal strength levels artificially above a set threshold, which will allow the data to be analyzed but resulting in no usable sequence.	</p>		<br>		<p class="main-text"><b>Noise.</b><br><br>Some background noise is always present in sequencing data. A high background and peaks under peaks (PUPS) characterizes noisy data. Figure 3 below shows an example of noisy data. The background is high enough to cause ambiguities in basecalling. Noise can be grouped into two categories:<br>		<ul>				<li>Noise throughout the sequence.</li>			<li>Noise up to or after a specific point in the sequence.</li>		</ul>	</p>		<br>		<p class="main-text">Possible causes for failed reactions are:<br>		<ul>				<li>Primer that has no annealing site, as in the figure above.</li>			<li>Insufficient template.</li>			<li>Contaminated template.</li>			<li>Insufficient primer.</li>			<li>Poor primer design.</li>		</ul>	</p>		<br>		<center>	<div class="screenshot">		<img src="../gfx/badsequence.png"><br>		<p>Figure 3. Example of a noisy sequence in 4Peaks.</p>	<div>	</center>		<br>		<p class="main-text">Not all noisy data is caused by low signal strengths, so other possibilities should be considered. Other potential causes for noise throughout the sequence:<br>		<ul>				<li>High signal strength, saturating the detector.</li>			<li>Contaminated template.</li>			<li>Multiple templates in the sequencing reaction.</li>			<li>Multiple priming sites.</li>			<li>Multiple primers.</li>			<li>No primer.</li>		</ul>	</p>		<br>	<p class="main-text"><b>Noise after a specific point in the sequence.</b><br><br>Noisy data after a specific point can have the following causes:		<ul>				<li>Mixed plasmid or PCR preparation, as in the example above.</li>			<li>Frame shift mutation.</li>			<li>A repeat region in the template.</li>			<li>Slippage after a homopolymer region in the template.</li>			<li>Primer-dimer contamination.</li>		</ul>	</p>		<br>		<p class="main-text"><b>Early Signal Loss.</b><br><br>Early signal loss can be grouped into two categories:		<ul>				<li>Gradual signal loss.</li>			<li>Abrupt signal loss.</li>		</ul>	</p>					<p class="main-text">Causes of gradual and abrupt signal losses could be one of the following:		<ul>				<li>Region of secondary structure in the template.</li>			<li>Template sequence idiosyncrasies.</li>			<li>Poor quantification of primer and/or template leading to top heavy data.</li>			<li>Salt contamination resulting in favored injection of lower molecular weight molecules leading to top heavy data.</li>		</ul>	</p>		<br>	<br>		<span class="section-header"><a name="reactions" id="reactions"><font color="#FF71A5">Troubleshooting DNA Sequencing Reactions</font></a></span>	<br>		<p class="main-text">Problems with DNA Sequencing reactions are most often due to:		<ul>				<li><a href="#template">Poor Template Quality.</a></li>			<br>			<li><a href="#primer">Primer-Related Problems.</a></li>			<br>			<li><a href="#dye">Excess Dye Peaks.</a></li>			<br>			<li><a href="#quantity">Quantity-Related Problems.</a></li>		</ul>	</p>	<br>		<p class="main-text"><a name="template" id="template"><b>Poor Template Quality.</b></a><br><br>Poor template quality is one of the most frequent causes of sequencing reaction problems. The quality of template DNA can be affected by the following:<br>		<ul>			<li>Contaminants: Residual salts or organic chemicals carried over from template preparation.</li>			<li>Incomplete removal of cellular components such as RNA, proteins, polysaccharides, and contaminating chromosomal DNA.</li>			<li>DNA degradation in storage.</li>			<li>Multiple DNA templates in the sequencing reaction.</li>		</ul>	</p>			<p class="main-text">		<b>Contaminants</b><br>		The presence of contaminants has a major influence on the quality of sequencing data obtained from the Hartwell Center. 		Proteins, RNA, chromosomal DNA, excess PCR primers, dNTPs, enzymes, buffer components, residual salts, organic 		chemicals, and detergents are all considered potential contaminants for DNA sequencing. In general, the presence of 		contaminants in template preparation can inhibit the sequencing reaction resulting in weak signal strengths. Weak signal 		strengths may or may not be accompanied by significant background noise.<br>		<br>		<b>Presence of EDTA</b><br>		The presence of EDTA in sequencing reactions can also affect the ability to obtain good quality sequencing data. The EDTA 		concentration present in a sample directly affects the sequencing signal strengths. The table below represents the 		correlation between EDTA concentration and the percent sequencing signal strength. In general, a higher concentration of 		EDTA decreases the sequencing signal strength, which will result in less accurate base-calling. EDTA will chelate the 		cofactors necessary for the DNA polymerase to work properly. Therefore, we recommend that you avoid EDTA.<br>		<br>	</p>		<center>	<table cellpadding="0" cellspacing="0">			<tr>				<td width="200"><center>EDTA Concentration</center></td>			<td align="center" bgcolor="#c0c0c0" width="2"> </td>			<td width="200"><center>Signal Strength</center></td>		</tr>		<tr>				<td><center>0 mM</center></td>			<td align="center" bgcolor="#c0c0c0" width="2"> </td>			<td><center>100%</center></td>		</tr>		<tr>				<td><center>0.1 mM</center></td>			<td align="center" bgcolor="#c0c0c0" width="2"> </td>			<td><center>94%</center></td>		</tr>		<tr>				<td><center>0.25 mM</center></td>			<td align="center" bgcolor="#c0c0c0" width="2"> </td>			<td><center>88%</center></td>		</tr>		<tr>				<td><center>0.5 mM</center></td>			<td align="center" bgcolor="#c0c0c0" width="2"> </td>			<td><center>25%</center></td>		</tr>		<tr>				<td><center>1.0 mM</center></td>			<td align="center" bgcolor="#c0c0c0" width="2"> </td>			<td><center>24%</center></td></tr>	</table>	</center>			<p class="main-text">		<b>DNA degradation in storage</b><br>		Repeat freeze-thaw cycles or nuclease contamination in a template preparation can cause DNA to degrade over time. <br>		<br>		<b>Multiple DNA templates in the sequencing reaction</b><br>		The presence of more than one template in a sequencing reaction will result in multiple overlapping sequences in the data. 		This can happen with both PCR products and cloned DNA templates. If PCR conditions are not optimized for complete 		specificity, more than one product can be produced. Most clean up procedures for PCR reactions are designed to remove 		unincorporated nucleotides, salts, and residual PCR primers, not secondary PCR products. Secondary PCR products can be 		distinguished by agarose gel electrophoresis. The band of interest can be gel purified and then resequenced.<br>	</p>		<br>		<p class="main-text"><a name="primer" id="primer"><b>Primer-Related Problems.</b></a><br><br>Primer related problems can be divided into four categories based on the type of problem seen in the data:<br>		<ul>			<li>No Recognizable Sequence.</li>			<li>Very Weak Sequence.</li>			<li>Two or More Sequences Present.</li>			<li>Primer-Dimers.</li>		</ul>	</p>			<p class="main-text">		<b>No Recognizable Sequence</b><br>		If there is no priming site for the primer in the template, no sequence data will be obtained. The data will show a flat line 		except for excess terminator peaks for dye terminator reactions. If the wrong primer is used for a particular vector or if 		a mutation is present in the primer binding site in the vector, the data would show a flat line. Figure 4 shows analyzed 		data from a reaction that had very low signal strength due to the absence of a priming site in the template for the primer 		that was used.	</p>		<br>		<center>	<div class="screenshot">		<img src="../gfx/wrongprimer.png"><br>		<p>Figure 4. Example of a sequence with the wrong primer in 4Peaks.</p>	<div>	</center>		<br>		<p class="main-text">		<b>Very Weak Sequence</b><br>		The signal strength in a sequencing reaction can be very weak if the primer anneals poorly because of a low melting 		temperature (Tm). Primers should have a Tm greater than 45&deg;C.	</p>			<p class="main-text">		<b>Two or More Sequences Present.</b><br>		More than one sequence can be present in the electropherogram if:<br>		<ul>				<li>The primer used in the reaction is contaminated with N-1 primer (a primer that is one base shorter than the desired primer).</li>			<li>Multiple primers present in the sequencing reaction.</li>			<li>There is a secondary hybridization site for the primer in the template.</li>			<li>In each of the above cases, there will be two or more sequences in the electropherogram. The data will look similar to that seen below with peaks under peaks except where the bases in both sequences are the same..</li>		</ul>	</p>			<p class="main-text">		<b>Multiple Primers.</b><br>		The presence of more than one primer in a sequencing reaction can be a problem when sequencing PCR products. Failure to 		completely remove the unincorporated primer from the PCR products will result in the carry over of some of these 		primers into the sequencing reaction. In Figure 5 you can see an example of a sequence generated with 2 primers. Almost each 		base has 2 peaks.	</p>		<br>		<center>	<div class="screenshot">		<img src="../gfx/2primers.png"><br>		<p>Figure 5. Example of a sequence with two primers in 4Peaks.</p>	<div>	</center>		<br>		<p class="main-text">		With dye terminator chemistries, the PCR extension products from the residual primers will also be labeled, which will 		result in a second sequence being present in the data. Complete removal of the unincorporated primers from the PCR 		amplification before sequencing will prevent this. Ultrafiltration using a Centricon-100 column will remove the 		unincorporated primers as well as the unused dNTPs.<br>	</p>		<p class="main-text">		<b>Secondary Hybridization.</b><br>		If a second hybridization site for the primer is present in the template, two sequences will be observed, resulting in noisy		 data. Hybridization at a secondary site that is not a perfect match for the primer can also happen at higher concentrations		  of primer. To avoid this, do not use primer amounts that exceed the recommended quantity of 3.2 pmoles.<br>		<br>		In cases where the secondary binding site is a perfect match for the primer, you will have to redesign the primer or use a 		different sequencing strategy.	</p>			<p class="main-text">		<b>Primer-Dimers.</b><br>		Primer-dimers are a result of intermolecular hybridization. These can occur between two primers with the same or 		different sequences. Usually this term is heard when referring to PCR amplifications where the forward and the reverse 		PCR primers hybridize generating a 25-60 base pair long product. This appears as a diffuse low molecular weight band 		on an agarose gel. Primer-dimers can also be generated between two primers with the same sequence as is some times seen 		in sequencing reactions. In both cases, the data looks the same.	</p>		<br>		<p class="main-text"><a name="dye" id="dye"><b>Excess Dye Peaks.</b></a><br><br>Excess dye peaks can be a common problem with dye terminator cycle sequencing chemistries. They are caused by incomplete removal of unincorporated, fluorescently labeled ddNTPs during post reaction cleanup. 	</p>			<p class="main-text">		<b>Second or Third Panel T Terminator Peak.</b><br>		A broad red peak is sometimes seen in the second or third panel of analyzed data with Big Dye terminator reactions. This is carryover contamination from unincorporated fluorsescent dyes. It is usually possible to call the bases manually under the red peak. See Figure 6 below.	</p>		<br>		<center>	<div class="screenshot">		<img src="../gfx/tpeak.gif"><br>		<p>Figure 6. Example a T peak causing a C to be seen as an N.</p>	<div>	</center>		<br>			<p class="main-text">		<b>First or Second Panel G Terminator Peak.</b><br>		A broad black peak is sometimes seen in the first and second panel of analyzed data. This is a result of the G dye being 		cleaved from the fluorescently labeled DNA in the purified sequencing reaction. It is not commonly seen on sequencing data 		produced by a good facility, but from time to time, it does occur. It is usually possible to manually call the bases 		under the black peak. In extreme instances, it can shorten the length of read because of "G dropout" or the loss of G signal 		further down the sequence.	</p>		<br>		<p class="main-text"><a name="quantity" id="quantity"><b>Quantity-Related Problems.</b></a><br><br>In order to obtain successful results with fluorescent based sequencing reactions the quantity of DNA template and primers is very important.	</p>			<p class="main-text">		<b>Not Enough Template.</b><br>		DNA template amounts that are lower than recommended can result in low, noisy signal with no well-defined peaks. In this 		case, the signal is usually too low to analyze. Data generated from the ABI 3700 DNA Analyzer has low background noise. 		Total signal strengths greater than 40 from this instrument can produce usable data. If you suspect you have used too little 		template, repeat the reaction and increase the amount of template.	</p>		<p class="main-text">		<b>Too Much Template.</b><br>		DNA template amounts that exceed the recommended amounts can also compromise DNA sequencing data. This is sometimes 		observed as pull-up peaks (PUPS), which are multiple peaks in the same position at certain points.<br>		<br>		Pull-up peaks are caused by very strong signals that saturate the detector. These extra peaks are observed at places in the 		electropherogram where there is a very tall peak, usually of one color. When the signals are very strong, more than one 		pull-up peak color can be observed.	</p>		<br>	<br>				<span class="section-header"><a name="composition" id="composition"><font color="#FF71A5">Troubleshooting DNA Sequence Composition Problems</font></a></span>	<br>		<p class="main-text">		The characteristics of DNA sequences can cause various types of issues:<br>		<ul>				<li><a href="#compression">Compressions.</a></li>			<br>			<li><a href="#gcrich">GC-Rich Templates (>70%) and Regions.</a></li>			<br>			<li><a href="#secondary">Secondary Structures.</a></li>			<br>			<li><a href="#gtrich">GT-Rich Templates.</a></li>			<br>			<li><a href="#homopolymer">Homopolymer Regions.</a></li>			<br>			<li><a href="#repetitive">Repetitive DNA.</a></li>		</ul>	</p>		<br>		<p class="main-text">		<a name="compression" id="compression"><b>Compressions.</b></a><br>		In DNA sequencing, band compressions are the result of the formation of secondary structures in the DNA fragments that 		are not eliminated by the denaturing conditions. The fragments do not migrate according to their size therefore, more than 		one fragment can migrate at the same position. With dye terminator chemistry, dITP is used in place of dGTP, which eliminates most compressions.	</p>		<br>		<p class="main-text">		<a name="gcrich" id="gcrich"><b>GC-Rich Templates and Regions.</b></a><br>		 The sequence composition of your template can also make obtaining good quality results difficult. Templates with greater		 than 70% GC content and small regions rich in GC content can be difficult to sequence. The most common problem seen 		 with templates and regions rich in GC content is weak signal strength. The addition of betaine (to a final concentration of 		 1M) to the cycle sequencing reaction can reduce the melting temperature of the template, which can facilitate successful 		 sequencing through these difficult areas.	</p>		<br>		<center>	<div class="screenshot">		<img src="../gfx/gcrich.png"><br>		<p>Figure 7. Example a decline in signal after a GC rich region in the template DNA in 4Peaks.</p>	<div>	</center>		<br>		<p class="main-text">		<a name="secondary" id="secondary"><b>Secondary structures.</b></a><br>		The ability to obtain good quality sequencing data can also be affected by the presence of secondary structure in the template 		strand. Regions of high secondary structure can affect the mobility of molecules, often resulting in difficulty obtaining good		 sequence data beyond this region.<br>		 <br> 		Ways to Obtain Sequence:<br> 		<ul>				<li>Use the same approaches as for <a href="#gcrich">GC-rich templates</a>.</li>			<li>Use a primer that anneals close to the region of signal loss.</li>			<li>Short insert libraries have used to solve problems with secondary structure in a genome-sequencing project.</li> 		</ul>	</p>		<br>		<p class="main-text">		<a name="gtrich" id="gtrich"><b>GT-Rich Templates.</b></a><br>		Figure 8 shows Big Dye terminator sequencing data where the signal dies after a GT-rich region. When the same template was sequenced using dRhodamine terminator chemistry, good data was generated beyond the GT-rich region.<br>		<br>		 Both chemistries use dITP instead of dGTP. Recall the dITP helps to eliminate compressions. The Big Dye terminator 		 chemistry uses dUTP instead of dTTP. dUTP is incorporated more efficiently thus generating a greater signal strength for T.		  Both dITP and dUTP will lower the melting temperature of the DNA. GT rich regions create a less stable duplex between the 		  extended primer and the template, so the polymerase has difficulty extending through this region.<br>		<br>		 Further studies with this template showed that the replacement of either dITP with dGTP or dUTP with dTTP allowed good 		 extension through this region (data not shown).<br>		 <br> 		Ways to Obtain Sequence:<br>		Addition of 1mM magnesium chloride to the reaction. This would increase the 		final concentration in the Big Dye terminator reaction from 2mM to 3mM 		magnesium chloride.	</p>		<br>		<center>	<div class="screenshot">		<img src="../gfx/gtrich.png"><br>		<p>Figure 8. Example a decline in signal after a GT rich region in the template DNA in 4Peaks.</p>	<div>	</center>		<br>		<p class="main-text">		<a name="homopolymer" id="homopolymer"><b>Homopolymer Regions.</b></a><br>		<b>Cloned DNA.</b><br>		Long homopolymer regions, consisting of multiple T's or A's in a row, can cause problems in DNA sequencing reactions due		 to 'slippage' in the region of the repeated base. The exact mechanism of slippage is not known, but presumed to be a result 		 of the two strands not pairing correctly during the denaturing and annealing processes of thermocycling conditions. 		 Sequence data can be clean through the homopolymer region, but often after that is noisy because of the presence of 		 multiple sequences.<br>		 <br>		Slippage events are length dependent and short homopolymer regions are rarely difficult to sequence. Slippage is more of a 		problem with T regions (A in the template strand) in Big Dye terminator because dUTP is used in the deoxynucleotide 		mixture.<br>		<br> 		Ways to Obtain Sequence:<br>		Use an anchored primer, i.e., a sequencing primer that is a poly T with an A, C, or G base at the 3« end. The 3« base anchors, 		or locks, the primer into place at the end of the homopolymer regions.<br>		<br>		Note: When sequencing with a poly T primer that has a "wobble" (A, C, or G at the 3« end) use 3.2 pmole of each primer, i.e., 		9.6 pmole total of primer.	</p>		<p class="main-text">		<b>PCR Templates.</b><br>		Slippage can also occur in PCR amplification and is often a problem when sequencing PCR products. It can occur in areas that		 are not normally problematic to sequence with cloned DNA. Presently, there is not an easy solution for the slippage 		 problem in PCR amplifications, but there are some approaches to try.<br>		<br> 		Ways to Obtain Sequence:<br>		<ol>				<li>Use an anchored primer, as described above. Sometimes they can be helpful in obtaining sequence data after homopolymer regions.</li>			<li>Try sequencing the complementary strand. This approach will give usable data up to the homopolymer region, but the same problem will occur afterwards.</li>			<li>Regardless, it can be very difficult to determine the exact number of bases present in a homopolymer region.</li>			<li>Clone the PCR product because in many cases, the cloned material will not show any evidence of slippage when sequenced. The disadvantage to this approach is the time and work involved in isolating and sequence the clone.</li>		</ol>	</p>		<br>		<p class="main-text">		<a name="repetitive" id="repetitive"><b>Repetitive DNA.</b></a><br>		Short repetitive regions of DNA (200 to 300 bases) are generally not difficult to sequence unless the base composition of 		the repeat is problematic. Repetitive regions of DNA consisting of more than 500 bases can make it difficult to acquire 		usable sequence data. Since there is no unique sequence in these repeats, synthesis of walking primers is not an option.<br>		 <br> 		Ways to Obtain Sequence:<br>		<ul>				<li>Use of directed deletions.</li>			<li>Use of an in vitro transposon system.</li>		</ul>	</p>		<br>		<p class="main-text">		Text adapted from a document originally published at: <a href="http://www.hartwellcenter.org/bio_services/dnaseqgeno/dnaseq/">http://www.hartwellcenter.org/bio_services/dnaseqgeno/dnaseq/</a><br>	</p>			<br>		<table border="0" width="100%">		<tr> 			<td bgcolor="#c0c0c0" height="3"></td>		</tr>	</table>	<br>	<table border="0" cellpadding="1" cellspacing="0" width="100%" class="menu">		<tr>			<td><center><a href="quickstart.html"><font size="2">Quick start</font></a></center></td>			<td><center><a href="main.html"><font size="2">Main Window</font></a></center></td>			<td><center><a href="file.html"><font size="2">File Menu</font></a></center></td>			<td><center><a href="edit.html"><font size="2">Edit Menu</font></a></center></td>			<td><center><a href="plugins.html"><font size="2">Plugins</font></a></center></td>		</tr>		<tr>			<td><center><a href="sequencing.html"><font size="2">Sequencing</font></a></center></td>			<td class="current"><center><a href="troubles.html"><font size="2">Troubleshooting</font></a></center></td>			<td><center><a href="history.html"><font size="2">Sequencing History</font></a></center></td>			<td><center><a href="nucleobytes.html"><font size="2">About Nucleobytes</font></a></center></td>			<td></td>		</tr>	</table>		<br><!-- AppleSegEnd --></body></html>